openedx/core/djangoapps/programs/tests/test_signals.py

```diff
@@ -224,3 +224,60 @@ def test_programs_enabled(self, mock_is_learner_issuance_enabled, mock_task):
         self.assertEqual(mock_is_learner_issuance_enabled.call_count, 1)
         self.assertEqual(mock_task.call_count, 1)
         self.assertEqual(mock_task.call_args[0], (TEST_USERNAME, TEST_COURSE_KEY))
+
+
+# The credentials app isn't installed for the CMS.
+@skip_unless_lms
+@mock.patch('openedx.core.djangoapps.programs.tasks.v1.tasks.update_certificate_visible_date_on_course_update.delay')
+@mock.patch(
+    'openedx.core.djangoapps.credentials.models.CredentialsApiConfig.is_learner_issuance_enabled',
+    new_callable=mock.PropertyMock,
+    return_value=False,
+)
+class CourseCertAvailableDateChangedReceiverTest(TestCase):
+    """
+    Tests for the `handle_course_cert_date_change` signal handler function.
+    """
+
+    @property
+    def signal_kwargs(self):
+        """
+        DRY helper.
+        """
+        return dict(
+            sender=self.__class__,
+            course_key=TEST_COURSE_KEY,
+        )
+
+    def test_signal_received(self, mock_is_learner_issuance_enabled, mock_task):  # pylint: disable=unused-argument
+        """
+        Ensures the receiver function is invoked when COURSE_CERT_DATE_CHANGE is
+        sent.
+
+        Suboptimal: because we cannot mock the receiver function itself (due
+        to the way django signals work), we mock a configuration call that is
```

Comment:
Please explain this statement in detail. It is possible to do manual signal connect in tests to do the receiver workflow

Reply:
It is complex to mock Django signals and their receivers. Instead, the concept used here is to test out the functionality of the receiver function to ensure that the receiver has been successfully executed. 
In our receiver function, we have a configuration call (`CredentialsApiConfig.current().is_learner_issuance_enabled`) which we are checking in our test to assert if the receiver was called successfully.